internal/controller/compaction/reconciler.go

```diff
@@ -493,3 +496,16 @@ func getCompactionJobArgs(etcd *druidv1alpha1.Etcd, metricsScrapeWaitDuration st
 
 	return command
 }
+
+func getEtcdCompactionAnnotations(etcdAnnotations map[string]string) map[string]string {
+	etcdCompactionAnnotations := make(map[string]string)
+
+	for key, value := range etcdAnnotations {
+		// Do not add annotation: `cluster-autoscaler.kubernetes.io/safe-to-evict: "false"` to compaction job's pod template.
+		if key == SafeToEvictKey && value == etcdAnnotations[SafeToEvictKey] {
```

Comment:
Don't `key == SafeToEvictKey` and `value == etcdAnnotations[SafeToEvictKey]` mean the same thing given the fact that `&&` is short-circuit?

`value` is nothing but `etcdAnnotations[key]`, which means:

when `key == SafeToEvictKey`
it is tautologically `value == etcdAnnotations[SafeToEvictKey]`
in the expression `key == SafeToEvictKey && value == etcdAnnotations[SafeToEvictKey]`.

```suggestion
		if key == SafeToEvictKey {
```

Reply:
addressed.

Reply:
Thanks.